src/sql/src/names.rs

```diff
@@ -1440,7 +1441,17 @@ impl<'a> NameResolver<'a> {
                         (full_name, item)
                     }
                 };
+
                 self.ids.insert(item.id());
+                self.ids.extend(
```

Comment:
This version only includes the types one level deep, whereas the previous version included the innermost, non container types. I have no idea which one is better.

Reply:
You're right--this should recursively find the inner elements. My oversight.

Reply:
Again, sorry for the late comment. It's not clear to me why we'd want to recurse here. For example if we have the following view definitions:

```SQL
CREATE VIEW v1 AS SELECT 1;
CREATE VIEW v2 AS SELECT * FROM v1;
CREATE VIEW v3 AS SELECT * FROM v2;

SELECT * FROM v3;
```

When planning the `SELECT` statement above, only `v3` will appear in the list of resolved IDs. We don't recurse and include `v1` and `v2`. So why do we treat types differently?